okhttp/src/main/java/com/squareup/okhttp/Request.java

```diff
@@ -233,8 +233,8 @@ public Builder method(String method, RequestBody body) {
       if (body != null && !HttpMethod.permitsRequestBody(method)) {
         throw new IllegalArgumentException("method " + method + " must not have a request body.");
       }
-      if (body == null && HttpMethod.permitsRequestBody(method)) {
-        body = RequestBody.create(null, Util.EMPTY_BYTE_ARRAY);
+      if (body == null && HttpMethod.requiresRequestBody(method)) {
+        throw new IllegalArgumentException("method " + method + " must have a request body.");
```

Comment:
Maybe we want a constant like RequestBody.EMPTY that is zero-bytes and has a null `Content-Type`?